### PR TITLE
Add initial docker and travis files

### PR DESCRIPTION
**Pull request by erikh360**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+env:
+  global:
+    - REGISTRY_USER=rapidproautomation
+    - IMAGE="praekeltfoundation/courier" VERSION="" REPO="nyaruka/courier"
+    - secure: TBwtks2Y5e4dSbV6FTnA5T3I178CC13K8mxmI2qk+eQ5kYRHOp4EqWVZF/jypPIoAy/YbO0CMB4Lhz3hDfilDE7pku6egGQT/mCT2UBCUGs+emIIWymBAbcx9uYnjwSRld5mrZ9lO59URNK6LIZd03IWi+urScRSI1k29qhN1PASZuUgXVpTNAVZNykXdM2MK6HxmyqRMpASMHhPATQnOvAspPRUgCkhxE5sG2Dr2baMqvR+019Wemj8JdZIIauNcgGEyCVXoDC9TAG0/abKxYpZvWJZ5UBboi3TMWgnKdlgBgdukh/pxWrtXxNKD6KpKR6oUuqqX7drbM78n4L74j4ytOFIv+JUEYTJb+mbSwGLaTdX2lnlop6y8E4FFjPp1JKtN43SdRS7XGB45MJoiyzeYFUPzdCED+p4rQQ4IKwxnF5IKYx5C/WXfY6bqaQR7PDaij8KbKSdD3R42NYjLdEFi1S5aLHvJVkmFQy/uxjpZ/Q2u4wgusOlixkBicAXHm3EuG61dhR0QuLUOEdwtR/thxWJxyK89T//8zWPyyrZv2WTDMOEXnAp8Av7Og4Cf816W6vjGE6GOfV/iWbeQ7L7jy9PvoDnaLiiDJEKhfUhOO+jrOqPyzhgkcfRXROe+y5CPAU5XvaQYaT5lsQtcRT9YX2PKkUuiER5phRm8DI=
+
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install -qy -o Dpkg::Options::="--force-confold" docker-ce coreutils
```
Comment (JayH5) [inline .travis.yml]: If you add `group: trusty_latest` you shouldn't need to update Docker.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  - LATEST_TAG="${LATEST_TAG//[v]}"
+  - VERSION=${VERSION:-$LATEST_TAG}
+
+install:
+  - echo "preventing default pip install here"
+
+script:
+  - docker pull debian:stretch-slim
+  - >
+    docker build --pull --cache-from "$IMAGE" \
```
Comment (JayH5) [inline .travis.yml]: If you want to use `--cache-from` you have to pull the image you are using. The `--pull` flag just pulls the base image (it's the same as the `docker pull` command above). I usually do a `docker pull "$IMAGE" || true` so the command succeeds even if the image hasn't been pushed yet.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - IMAGE="praekeltfoundation/courier" VERSION="" REPO="nyaruka/courier"
+    - secure: TBwtks2Y5e4dSbV6FTnA5T3I178CC13K8mxmI2qk+eQ5kYRHOp4EqWVZF/jypPIoAy/YbO0CMB4Lhz3hDfilDE7pku6egGQT/mCT2UBCUGs+emIIWymBAbcx9uYnjwSRld5mrZ9lO59URNK6LIZd03IWi+urScRSI1k29qhN1PASZuUgXVpTNAVZNykXdM2MK6HxmyqRMpASMHhPATQnOvAspPRUgCkhxE5sG2Dr2baMqvR+019Wemj8JdZIIauNcgGEyCVXoDC9TAG0/abKxYpZvWJZ5UBboi3TMWgnKdlgBgdukh/pxWrtXxNKD6KpKR6oUuqqX7drbM78n4L74j4ytOFIv+JUEYTJb+mbSwGLaTdX2lnlop6y8E4FFjPp1JKtN43SdRS7XGB45MJoiyzeYFUPzdCED+p4rQQ4IKwxnF5IKYx5C/WXfY6bqaQR7PDaij8KbKSdD3R42NYjLdEFi1S5aLHvJVkmFQy/uxjpZ/Q2u4wgusOlixkBicAXHm3EuG61dhR0QuLUOEdwtR/thxWJxyK89T//8zWPyyrZv2WTDMOEXnAp8Av7Og4Cf816W6vjGE6GOfV/iWbeQ7L7jy9PvoDnaLiiDJEKhfUhOO+jrOqPyzhgkcfRXROe+y5CPAU5XvaQYaT5lsQtcRT9YX2PKkUuiER5phRm8DI=
+
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install -qy -o Dpkg::Options::="--force-confold" docker-ce coreutils
+
+before_script:
+  - LATEST_TAG=$(curl -s https://api.github.com/repos/$REPO/tags  | jq -r '.[0].name')
+  - LATEST_TAG="${LATEST_TAG//[v]}"
```
Comment (JayH5) [inline .travis.yml]: You can include the `v` in the version if you like. docker-ci-deploy should handle it fine. It's just a matter of taste.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  - VERSION=${VERSION:-$LATEST_TAG}
+
+install:
```
Comment (JayH5) [inline .travis.yml]: You can set `install` to `null`/`~` or `[]` to have the same effect.

File: .travis.yml (hunk 1 above)
Comment (JayH5) [reply .travis.yml]: `coreutils` needs updating?

File: .travis.yml (hunk 3 above)
Comment (erikh360) [reply .travis.yml]: I had to remove it to pull the binary from github in the Dockerfile from this url:
https://github.com/nyaruka/courier/releases/download/v1.2.84/courier_1.2.84_linux_amd64.tar.gz

Do you think it's better to just do that in the Dockerfile?

Comment (JayH5) [reply .travis.yml]: Ah jeese so they have the v in one place and not in the other? :/ yeah then I think what you've got is best.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+before_script:
+  - LATEST_TAG=$(curl -s https://api.github.com/repos/$REPO/tags  | jq -r '.[0].name')
+  - LATEST_TAG="${LATEST_TAG//[v]}"
+  - VERSION=${VERSION:-$LATEST_TAG}
+
+install: []
+
+script:
+  - docker pull debian:stretch-slim
```
Comment (JayH5) [inline .travis.yml]: You shouldn't need this line since you're using `--pull` below.

**Review by JayH5 — APPROVED** [review]
:+1: although can you please make a follow-up PR to make courier run as a non-root user?

**Comment by erikh360** [comment]
Cool, I'll do that next. Thanks for your help so far @JayH5

**Pull request merged**